src/middle/libraryjs.sk

```diff
@@ -317,133 +321,133 @@ namespace List {
 
 namespace StringMap {
   def new StringMap<T> {
-    return __create(null)
+    return __createMap()
   }
 }
 
 class StringMap {
   def [](key string) T {
     assert(key in self)
-    return (self as dynamic)[key]
+    return (self as dynamic).get(key)
+  }
+
+  def []=(key string, value T) T {
+    (self as dynamic).set(key, value)
+    return value
   }
 
   def {...}(key string, value T) StringMap<T> {
     self[key] = value
     return self
   }
 
+  def in(key string) bool {
+    return (self as dynamic).has(key)
+  }
+
   def count int {
-    return keys.count
+    return (self as dynamic).size
   }
 
   def isEmpty bool {
-    for key in self as dynamic {
-      return false
-    }
-    return true
+    return count == 0
   }
 
   def get(key string, defaultValue T) T {
-    var value = (self as dynamic)[key]
+    const value = (self as dynamic).get(key)
     return value != dynamic.void(0) ? value : defaultValue # Compare against undefined so the key is only hashed once for speed
   }
 
   def keys List<string> {
-    return dynamic.Object.keys(self)
+    return dynamic.Array.from((self as dynamic).keys())
```

Comment:
`Map.keys` and `Map.values` each return iterators, so the `Array.from` is necessary to return arrays.